core/debug/src/debug.c

```diff
@@ -50,25 +50,31 @@ void debug_semihosting_enable(void)
 
 UVISOR_WEAK void default_putc(uint8_t data)
 {
-    if (DEBUG_SEMIHOSTING_MAGIC == g_semihosting_magic) {
```

Comment:
i suggest not to remove semi-hosting print logic - we still need it for debugging stuff during early boot stages.
we can duplicate print to two sinks - that is what all decent loggers do :)

Reply:
Not removed, divided into two separate checks.